### PR TITLE
Add basic charm functionality and utilities

### PR DESCRIPTION
**Pull request by kelkawi-a**
This PR adds basic functionality to the Airbyte UI charm:

- Adds OCI image and entry point to start the application
- Adds update status hook logic to start the application based on a liveness check
- Adds peer relation for sharing state across units
- Adds logging utility
- Adds restart action
- Updates README and contributing guidelines

Note: Some links point to pages in the Airbyte Charmhub page and server repo which may not exist yet.

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -43,6 +23,11 @@ bases:
       channel: "22.04"
 
 
+# Metadata
+peers:
+  peer:
+    interface: airbyte
```
Comment (mertalpt) [inline charmcraft.yaml:29]: question: What is the purpose of this peer relation?

I saw it in the other charm as well. If it is meant to be between the UI and the server, is it supposed to be a peer relation?

**Review by mertalpt — APPROVED** [review]
LGTM.

File: charmcraft.yaml (hunk 1 above)
Comment (kelkawi-a) [reply charmcraft.yaml:29]: The peer relation is the relation of the charm with itself, it provides a relation databag that allows us to store relevant data to be accessed by the application's units later on (e.g. db connection).

Comment (mertalpt) [reply charmcraft.yaml:29]: I see. I was confused by both relations being named `airbyte`.

File: src/charm.py [hunk 2]
```diff
@@ -1,103 +1,142 @@
...
+class AirbyteK8SOperatorCharm(CharmBase):
+    """Charm the application."""
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on['httpbin'].pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self._state = State(self.app, lambda: self.model.get_relation("peer"))
+
+        self.name = "airbyte-webapp"
```
Comment (AmberCharitos) [inline src/charm.py:28]: perhaps: `self.app.name`

File: src/charm.py [hunk 3]
```diff
@@ -1,103 +1,142 @@
...
+            event:The event triggered by the restart action
+        """
+        container = self.unit.get_container(self.name)
+        if not container.can_connect():
+            event.defer()
+            return
+
+        self.unit.status = MaintenanceStatus("restarting application")
+        container.restart(self.name)
+        self.unit.status = ActiveStatus()
```
Comment (AmberCharitos) [inline src/charm.py:86]: I suppose this `ActiveStatus` will be set in the next successful `update_status` - Is there any reason it needs to be set here immediately following restart?

File: src/charm.py [hunk 4]
```diff
@@ -1,103 +1,142 @@
...
+        # TODO (kelkawi-a): update this to get server application name through charm relation
+        context = {
+            "API_URL": "/api/v1/",
+            "AIRBYTE_EDITION": "community",
+            "INTERNAL_API_HOST": "airbyte-k8s:8001",
+            "CONNECTOR_BUILDER_API_HOST": "airbyte-k8s:80",
+            "KEYCLOAK_INTERNAL_HOST": "localhost",
+        }
+
+        self.model.unit.set_ports(8080)
```
Comment (AmberCharitos) [inline src/charm.py:107]: `WEB_UI_PORT` could be used here instead of 8080

File: src/charm.py [hunk 5]
```diff
@@ -1,103 +1,142 @@
...
+
+        pebble_layer = {
+            "summary": "airbyte layer",
             "services": {
-                "httpbin": {
-                    "override": "replace",
-                    "summary": "httpbin",
-                    "command": "gunicorn -b 0.0.0.0:80 httpbin:app -k gevent",
+                self.name: {
+                    "summary": "airbyte webapp",
```
Comment (AmberCharitos) [inline src/charm.py:117]: `self.app.name ` ? to avoid having more literals to change

**Review by AmberCharitos — APPROVED** [review]
LGTM some minor nitpicks

File: src/charm.py (hunk 2 above)
Comment (kelkawi-a) [reply src/charm.py:28]: `self.app.name` is the name that the user has deployed the application with, which could vary. Fixing the name of the container to `airbye-webapp` would be more suitable to have uniform instructions on accessing the container.

**Pull request merged**